db/migrate/20191126181009_seed_test_results.rb

```diff
@@ -13,24 +15,23 @@ def up
       rule_results_by_end_time = scan_results(host, profile)
         .group_by(&:end_time)
       rule_results_by_end_time.each do |end_time, rule_results|
-        test_result = TestResult.create!(
-          host: host,
-          profile: profile
+        rules_passed = rule_results.count { |rr| RuleResult::PASSED.include?(rr.result) }
+        rules_failed = rule_results.count { |rr| RuleResult::FAIL.include?(rr.result) }
+        rule_results_set << rule_results
+        test_results << TestResult.new(
+          host_id: host.id,
+          profile_id: profile.id,
+          start_time: rule_results[0].start_time || Time.now.utc,
+          end_time: rule_results[0].end_time || Time.now.utc,
+          score: compliance_score(rules_passed, rules_failed)
```

Comment:
This method doesn't protect against `ZeroDivisionError`, and I don't see any checks. If a user scanned a system against a profile from a benchmark of a different OS, it will have 0 rule results.